indico/modules/events/contributions/controllers.py

```diff
@@ -95,8 +78,40 @@ class RHContributions(RHManageContributionsBase):
     """Display contributions management page"""
 
     def _process(self):
+        contrib_report_config = reporter.get_report_config(self.event_new.id)
+        if 'config' in request.args:
+            return redirect(url_for('.manage_contributions', self.event_new))
+
+        contrib_report_args = reporter.get_contrib_report_args(self.event_new, contrib_report_config['filters'])
+
         return WPManageContributions.render_template('management/contributions.html', self._conf, event=self.event_new,
-                                                     **_get_contribution_list_args(self.event_new))
+                                                     **contrib_report_args)
+
+
+class RHContributionsListCustomize(RHManageContributionsBase):
+    """Filter options for the contribution list of an event"""
+
+    def _process_GET(self):
+        contrib_report_config = reporter.get_report_config(self.event_new.id)
+        filterable_items = reporter.FILTERABLE_ITEMS
+        filterable_items['session']['filter_choices'] = {unicode(s.id): s.title
```

Comment:
Because when we read the filters from the request [1]
```
options = request.form.getlist('field_{}'.format(item_id))
```
the `options` will be a list of strings. I could convert it to `int` here but no-one guarantees that the list will contain only numerical ids.
In the template [2] I am using the following line
```
{% if value in filters[item_id] %}
```
to initialize the checkboxes. So if the `value` is an `int` and the `filters[item_id]` is a list of strings, then the comparison evaluates to `False`.

[1] https://github.com/nop33/indico/blob/wip/contr-filter/indico/util/reporter.py#L66 
[2] https://github.com/nop33/indico/blob/wip/contr-filter/indico/modules/events/contributions/templates/management/contrib_report_filter.html#L17